src/main/java/io/github/minecraftcursedlegacy/api/client/AtlasMap.java

```diff
@@ -0,0 +1,169 @@
+package io.github.minecraftcursedlegacy.api.client;
+
+import java.util.OptionalInt;
+
+import net.minecraft.client.texture.TextureManager;
+import net.minecraft.item.ItemInstance;
+import net.minecraft.item.ItemType;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+import io.github.minecraftcursedlegacy.impl.client.AtlasMapper;
+
+/**
+ * Utility methods for {@link ItemType}s which use textures from other than the main item atlas.
+ *
+ * @author Chocohead
+ */
+public final class AtlasMap {
+	private AtlasMap() {
+	}
+
+	/**
+	 * Register the given atlas for use with the {@link ItemType} corresponding to the given 
+	 * {@link ItemInstance#itemId} when it has the given {@link ItemInstance#getDamage()}.
+	 *
+	 * @param stack The item type and damage which uses the given atlas
+	 * @param atlas The location of the custom atlas
+	 *
+	 * @throws NullPointerException If atlas is <tt>null</tt>
+	 */
+	public static void registerAtlas(ItemInstance stack, String atlas) {
+		registerAtlas(stack.itemId, stack.getDamage(), atlas);
+	}
+
+	/**
+	 * Register the given atlas for use with the given {@link ItemType},
+	 * regardless of what {@link ItemInstance#getDamage()} on an instance of it returns.
+	 *
+	 * @param item The item which uses the given atlas
+	 * @param atlas The location of the custom atlas
+	 *
+	 * @throws NullPointerException If atlas is <tt>null</tt>
+	 */
+	public static void registerAtlas(ItemType item, String atlas) {
+		registerAtlas(item.id, atlas);
+	}
+
+	/**
+	 * Register the given atlas for use with the {@link ItemType} corresponding to the given ID
+	 * when {@link ItemInstance#getDamage()} on an instance of it is the same as the given meta.
+	 *
+	 * @param itemID The ID of the item which uses the given atlas
+	 * @param meta The damage value an instance must have to use this atlas
+	 * @param atlas The location of the custom atlas
```

Comment:
can we have full stops at the end of each javadoc :wink:

Reply:
Do you just want it on the `@param`s or on the `@return`/`@throws` too?

Reply:
on return and throws too